### PR TITLE
fixes zombie, SBG, alk persp

### PR DESCRIPTION
**Pull request by zeskorion**
## About The Pull Request

zombie disease now creates the organ itself instead of directly synthesizing romerol. 
makes the death roaches symptom on SBG actually work
ports the following fix: https://github.com/tgstation/tgstation/pull/46063

## Why It's Good For The Game

1. romerol doesnt work if added directly through the reagent adding proc
2. this uses organ/zombie infection instead of organ/zombie infection/nodamage, meaning viro can't have the insurance policy of romerol without slowly dying if they inject themselves with this shit. (though, this symptom kills fast enough that that should not be a problem)
3. alk persp doesnt actually work right
4. fixes a threshold

## Changelog
:cl:
fix: zombievirus works properly now. for the record, the part that was broken was the shit that was there. not my own code. thanks for the help, mark
fix: SBG death roaches work properly now
fix: alkali perspiration actually works as advertised now
code: adds OnDeath proc for diseases
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
While you're at it, could you update the description of autophagocytosis necrosis on the wiki and in-game to reflect the fact that the zombie threshold effect is obtainable?

**Comment by ATH1909** [comment]
You may or may not also want to make autophagocytosis not damage/inject chems into the undead, so that virologists can combo this symptom with necrotic metabolism without killing their new zombies.

**Comment by zeskorion** [comment]
> While you're at it, could you update the description of autophagocytosis necrosis on the wiki and in-game to reflect the fact that the zombie threshold effect is obtainable?

1. Irs supposed to be somewhat sekrit , but the ingame text has *hints* as to the function 
2. An old symptom to assist in this purpose is being readded

**Comment by zeskorion** [comment]
fucking finally.

File: code/datums/diseases/advance/symptoms/cockroach.dm [hunk 1]
```diff
@@ -40,9 +40,18 @@
...
 			if(prob(50))
 				to_chat(M, "<span class='notice'>You feel something crawling in your pipes!</span>")
+	
+/datum/symptom/cockroach/OnDeath(datum/disease/advance/A)
+	if(!..())
+		return
+	if(death_roaches)
+		var/mob/living/M = A.affected_mob
+		to_chat(M, "<span class='warning'>Your pores explode into a colony of roaches!</span>")
+		for(var/i in 1 to rand(1,5))new /mob/living/simple_animal/cockroach(M.loc)
```
Comment (ike709) [inline code/datums/diseases/advance/symptoms/cockroach.dm]: Christ.

File: code/datums/diseases/advance/symptoms/symptoms.dm [hunk 2]
```diff
@@ -74,7 +74,10 @@
 
 /datum/symptom/proc/generate_threshold_desc()
 	return
-
+	
+/datum/symptom/proc/OnDeath(datum/disease/advance/A)
+	return !neutered
```
Comment (ike709) [inline code/datums/diseases/advance/symptoms/symptoms.dm:79]: I don't understand viro code very well. Why are you returning `!neutered`? Shouldn't it just be `neutered`?

File: code/datums/diseases/advance/symptoms/cockroach.dm (hunk 1 above)
Comment (zeskorion) [reply code/datums/diseases/advance/symptoms/cockroach.dm]: ?

Comment (zeskorion) [reply code/datums/diseases/advance/symptoms/cockroach.dm]: if it's the last line you're worried about, that's literally the only way it could pass travis

Comment (ike709) [reply code/datums/diseases/advance/symptoms/cockroach.dm]: You put two lines of code on one line.

File: code/datums/diseases/advance/symptoms/symptoms.dm (hunk 2 above)
Comment (zeskorion) [reply code/datums/diseases/advance/symptoms/symptoms.dm:79]: I am unsure. the code i used here is some shit from an older /tg/ symptom pr that went stale for having "fetish content" (fart symptoms). As i understand, it only returns if the symptom is not neutered.

Comment (ike709) [reply code/datums/diseases/advance/symptoms/symptoms.dm:79]: No, it returns the _opposite_ of the neutered value. So if neutered is false, it returns true. And vice-versa.

File: code/datums/diseases/advance/symptoms/cockroach.dm (hunk 1 above)
Comment (zeskorion) [reply code/datums/diseases/advance/symptoms/cockroach.dm]: only thing that travis didnt hate. i mean, ill try again...

Comment (zeskorion) [reply code/datums/diseases/advance/symptoms/cockroach.dm]: .... what the fuck, it's fine now?


some shit i'll never understand

**Pull request merged**